### DISCUSSION TITLE
How to grab and view the log of fwupd?

### DISCUSSION THREAD

**Discussion by airsoup** [fwupd/fwupd#5952] [Q&A] [answered]
Hi,
I encountered some issues while updating firmware on fwupd and would like to check the log, such as "g_debug" in the code.
Do I need to enable debug mode for debugging information? And how to obtain log messages?

**Comment by hughsie** [comment, marked as answer]
If you run `fwupd` (the daemon) or `fwupdtool` (the in-process tool) as root you need to use `-v` to get one level of debugging, and `-v` again to get another layer. So, for instance, you can do something like `sudo fwupdtool get-devices --verbose --verbose` to get *everything*.

**Reply by airsoup** [reply]
Okay, adding the `-vv` parameter can display `g_debug` messages. Thank you for your reply and have a pleasant time.
